op-node/rollup/driver/state.go

```diff
@@ -508,7 +509,8 @@ func (s *SyncDeriver) onStepEvent() {
 		s.Log.Error("Derivation process error", "err", err)
 		s.Emitter.Emit(StepReqEvent{})
 	} else {
-		s.Emitter.Emit(StepReqEvent{ResetBackoff: true}) // continue with the next step if we can
+		// Revisit SyncStep in 1/2 of a L2 block.
+		s.Emitter.Emit(StepDelayedReqEvent{Delay: (time.Duration(s.Config.BlockTime) * time.Second) / 2})
```

Comment:
So we don't have an error, haven't reached an idle state so there's more to do, but we want to delay.  Is that because there's async stuff happening like importing the next block?  I'm just nervous that a delay here could have major impacts to sync times if the node is behind.

Reply:
The main issue is that it needs to poll/retry L1 traversal during regular operation. For this we need to trigger a re-attempt, either by monitoring L1 proactively, or by making the derivation-pipeline do that. Hence the delay here.


If the derivation-pipeline is busy, then it will emit an event that things can continue straight away (i.e. there's more data): https://github.com/ethereum-optimism/optimism/blob/793f98d851489be8257eb811f2466bd2370f4175/op-node/rollup/derive/deriver.go#L124
And a non-delayed step will then be requested.

Ideally we clean up the Driver and SyncDeriver more; it can just be a simple event handler that ties together some actions (like more-data => step) but not do anything that's not triggered by an event or outside input.